### DISCUSSION TITLE
VQ-VAE LDM tutorial

### DISCUSSION THREAD

**Discussion by guidodf09** [Project-MONAI/GenerativeModels#496] [Q&A]
Hi everyone,

I am trying to test the 2D LDM tutorial with the AEKL, as in https://github.com/Project-MONAI/GenerativeModels/blob/main/tutorials/generative/2d_ldm/2d_ldm_tutorial.ipynb . I was wondering if there is an available version of this tutorial (basically just the U-net training part of the LDM) using MONAI's VQ-VAE, as I have not been able to find one. Thanks in advance. 

Guido

**Comment by marksgraham** [comment]
There isn't but the VQ-VAE and AE can be used very similarly so you should be able to just swap out the AE for a VQ-VAE, maybe using [this](https://github.com/Project-MONAI/GenerativeModels/blob/main/tutorials/generative/2d_vqvae/2d_vqvae_tutorial.ipynb) for guidance on how to use the VQ-VAE if needed. 

I also have a research codebase that makes use of MONAI gen to train an LDM with a VQ-VAE which may be useful [here](https://github.com/marksgraham/ddpm-ood#run-with-ldm)
